openghg/store/base/_base.py

```diff
@@ -509,16 +538,18 @@ def _validate_datasources(
         """
         validate_params = self.find_data_schema_inputs()
 
+        if filepath is not None:
```

Comment:
If the `filepath` input was already `Path | list[Path] | None` we wouldn't need this.